code/procs/gamehelpers.dm

```diff
@@ -452,6 +452,28 @@ var/obj/item/dummy/click_dummy = new
 				. += R
 				break
 
+/proc/get_nonvirtual_areas_with_turfs(var/areatype, var/list/blacklist = null) // for spy thief!
```

Comment:
not liking this copy paste too much, why not just wrap a filter around get_areas_with_turfs
also god that code is pretty gross, like I get what it's doing but it's gross and it actually has a bug in it (null entries in R.contents I think)

Reply:
I wanted to limit the function to iterating through all the areas once, and I wanted to leave the original function intact. I agree it's pretty gross; this seemed like the least painful way to solve this issue without introducing new bugs

Reply:
You can achieve the blacklist with just a wrapper though, just check if the passed areatype string is in the blacklist, else call nonvirtual, also this wouldn't work anyways because you're comparing strings to area types via istype

Reply:
Areatype is passed an area by spy_theft, not a string. I don't think your wrapper solution would work. Consider getting passed the areatype `/area/station`. This isn't on the blacklist, but `"Storage Area` (which is the name of the area type `/area/station/storage`) is on the blacklist. Perhaps there's a better solution using `filtered_concrete_typesof` to get all the area types except those on the blacklist, but that doesn't quite cover the `virtual` case (there are `virtual` subtypes of non-virtual types).